wgpu-hal/src/vulkan/mod.rs

```diff
@@ -355,6 +355,11 @@ struct Swapchain {
     /// index as the image index, but we need to specify the semaphore as an argument
     /// to the acquire_next_image function which is what tells us which image to use.
     next_semaphore_index: usize,
+    /// The times which will be set in the next present times.
+    ///
+    /// SAFETY: This is only set if [wgt::Features::VULKAN_GOOGLE_DISPLAY_TIMING] is enabled, and
+    /// so `VK_GOOGLE_display_timing` is set.
```

Comment:
"is set" means "Vulkan extension is enabled"?

Reply:
Already changed the wording here - how is the new one.

Reply:
Aynchronously writing reviews while a PR is actively force-pushed is very hard, yes - let me slowly close/rephrase whatever has already been changed since posting.

In hindsight I'm not sure about having `SAFETY` docs on a field in the first place: there's nothing unsafe happening on this line. Unwrapping on `Option` isn't unsafe either?

Reply:
> In hindsight I'm not sure about having `SAFETY` docs on a field in the first place: there's nothing unsafe happening on this line. Unwrapping on `Option` isn't unsafe either?

The field's value is relied upon for safety. There is a debug assertion protecting this condition.

However, it is UB in Vulkan to add the Present information without the extension enabled. We certainly could change the comment to not use the "# Safety" format, if that is preferable